rmw_zenoh_cpp/src/detail/rmw_data_types.cpp

```diff
@@ -522,6 +552,20 @@ void client_data_handler(z_owned_reply_t * reply, void * data)
     );
     return;
   }
+
```

Comment:
This decrement should be done in the `drop` function of the `z_owned_closure_reply_t`. There is no guarantee that the `call` function will be called only once per query, while `drop` will be called once per query at the end.